comms/core/src/connectivity/proactive_dialer.rs

```diff
@@ -0,0 +1,382 @@
+                "({}) No healthy peer candidates available for proactive dialing",
+                task_id
+            );
+            proactive_dialing_metrics::observe_proactive_dialing_execution_time(start_time.elapsed());
+            return Ok(0);
+        }
+
+        // Update candidate metrics
+        proactive_dialing_metrics::set_available_peer_candidates(candidates.len());
+
+        // Initiate concurrent dials
+        let dialed_count = self.dial_peers_concurrently(candidates, task_id).await;
+
+        info!(
+            target: LOG_TARGET,
+            "({}) Proactive dialing initiated for {} peers ({} needed connections)",
+            task_id,
+            dialed_count,
+            needed
+        );
+
+        // Record final metrics
+        proactive_dialing_metrics::observe_proactive_dialing_execution_time(start_time.elapsed());
+
+        Ok(dialed_count)
+    }
+
+    /// Calculate recent connection success rate across all peers
+    fn calculate_recent_success_rate(
+        &self,
+        connection_stats: &std::collections::HashMap<NodeId, super::connection_stats::PeerConnectionStats>,
+    ) -> f32 {
+        if connection_stats.is_empty() {
+            return 0.25; // Conservative Bayesian prior default
+        }
+
+        let window = self.config.success_rate_tracking_window;
+        let total_stats: Vec<f32> = connection_stats
+            .values()
+            .map(|stats| stats.success_rate(window))
+            .collect();
+
+        if total_stats.is_empty() {
+            return 0.25; // Conservative Bayesian prior default
+        }
+
+        let average = total_stats.iter().sum::<f32>() / total_stats.len() as f32;
+        average.clamp(0.1, 1.0) // Clamp between 10% and 100%
+    }
+
+    /// Calculate how many peers to dial based on needed connections and success rate
+    fn calculate_dial_count(&self, needed: usize, success_rate: f32) -> usize {
+        let base_count = needed as f32 * self.config.dialing_multiplier;
+        let adjusted_count = base_count / success_rate.max(0.1); // Prevent division by very small numbers
+
+        #[allow(clippy::cast_possible_truncation)]
+        let final_count = adjusted_count.ceil() as usize;
+
+        // Cap the dial count to prevent overwhelming the network
+        const MAX_CONCURRENT_DIALS: usize = 20;
+        final_count.max(needed).min(MAX_CONCURRENT_DIALS)
+    }
+
+    /// Select healthy peer candidates for dialing
+    async fn select_healthy_dial_candidates(
+        &self,
+        pool: &ConnectionPool,
+        connection_stats: &std::collections::HashMap<NodeId, super::connection_stats::PeerConnectionStats>,
+        count: usize,
+        task_id: u64,
+    ) -> Result<Vec<Peer>, ConnectivityError> {
+        // Get all known communication peers
+        let all_peers = self
+            .peer_manager
+            .all(None)
+            .await
+            .map_err(ConnectivityError::PeerManagerError)?;
```

Comment:
```suggestion
        let all_peers = self
            .peer_manager
            .()
            .await
            .map_err(ConnectivityError::PeerManagerError)?;
```
Dont use all, use the SQL methods, or create new ones